### PR TITLE
Implemented logic to apply threshold rule in threshold management page if jobId is passed in route parameter(#2mky6cn)

### PR DESCRIPTION
**Pull request by disha1202**
### Related Issues
 

 Closes #

 ### Short Description and Why It's Useful
 

 ### Screenshots of Visual Changes before/after (If There Are Any)
 

 **IMPORTANT NOTICE** - Remember to add changelog entry

 ### Contribution and Currently Important Rules Acceptance
 

 - [ ] I read and followed [contribution rules](https://github.com/hotwax/threshold-management#contribution-guideline)

### CODE REVIEW

File: src/store/modules/job/getters.ts [hunk 1]
```diff
@@ -39,6 +40,30 @@ const getters: GetterTree <JobState, RootState> = {
...
+      const operator = tagsIncluded.indexOf('AND') ? 'AND' : 'OR'
+      tags = tags.map((x: any) => {
+        return JSON.parse(x[0])
+      });
+      return { tags, operator }
+    },
+    getExcludedTagsAndOperator: (state) => (id: string): any => {
+      const thresholdRule = state.thresholdRules[id];
+      if (!thresholdRule) return "";
+      const tagsIncluded = thresholdRule.json.filter.find((filter: any) => filter.startsWith("-tags:"))
```
Comment (ymaheshwari1) [inline src/store/modules/job/getters.ts:58]:
```suggestion
      const tagsExcluded = thresholdRule.json.filter.find((filter: any) => filter.startsWith("-tags:"))
```

File: src/store/modules/job/getters.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import { GetterTree } from 'vuex'
 import JobState from './JobState'
 import RootState from '../../RootState'
+import  parser from 'boolean-parser'
```
Comment (ymaheshwari1) [inline src/store/modules/job/getters.ts:4]:
```suggestion
import parser from 'boolean-parser'
```

File: src/store/modules/job/getters.ts [hunk 3]
```diff
@@ -39,6 +40,30 @@ const getters: GetterTree <JobState, RootState> = {
...
+    },
+    getExcludedTagsAndOperator: (state) => (id: string): any => {
+      const thresholdRule = state.thresholdRules[id];
+      if (!thresholdRule) return "";
+      const tagsIncluded = thresholdRule.json.filter.find((filter: any) => filter.startsWith("-tags:"))
+      if (!tagsIncluded) return ""
+      let tags = parser.parseBooleanQuery(tagsIncluded.substring(tagsIncluded.indexOf(":") + 1).trim());
+      const operator = tagsIncluded.indexOf('AND') ? 'AND' : 'OR'
+      tags = tags.map((x: any) => {
+        return JSON.parse(x[0])
```
Comment (ymaheshwari1) [inline src/store/modules/job/getters.ts:63]: We don't need return here, we can directly write JSON.parse(x[0])

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/store/modules/job/getters.ts (hunk 1 above)
Comment (disha1202) [reply src/store/modules/job/getters.ts]: Improved variable name.

File: src/store/modules/job/getters.ts (hunk 2 above)
Comment (disha1202) [reply src/store/modules/job/getters.ts]: Improved spacing

File: src/store/modules/job/getters.ts (hunk 3 above)
Comment (disha1202) [reply src/store/modules/job/getters.ts]: Made the requested change.

File: src/store/modules/job/getters.ts [hunk 4]
```diff
@@ -39,21 +40,29 @@ const getters: GetterTree <JobState, RootState> = {
...
+    getTags: (state) => (id: string, type: string): any => {
       const thresholdRule = state.thresholdRules[id];
       if (!thresholdRule) return "";
-      const tagsExcluded = thresholdRule.json.filter.find((filter: any) => filter.startsWith("-tags:"))
-      if (!tagsExcluded) return ""
-      return tagsExcluded.substring(tagsExcluded.indexOf(":") + 1)
-    },
+      let tags;
+      if(type === 'included'){
+        tags = thresholdRule.json.filter.find((filter: any) => filter.startsWith("tags:"))
```
Comment (ymaheshwari1) [inline src/store/modules/job/getters.ts:59]:
```suggestion
        tags = thresholdRule.json.filter.find((filter: any) => filter.startsWith(`type === 'included' ? tags: : -tags:`)) 
```

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/services/JobService.ts [hunk 5]
```diff
@@ -23,6 +23,15 @@ const updateJob = async (payload: any): Promise <any>  => {
   });
 }
 
+const updateRuntimeData = async (payload: any): Promise <any>  => {
```
Comment (ymaheshwari1) [inline src/services/JobService.ts:26]: Remove the unwanted service.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/SelectProduct.vue [hunk 6]
```diff
@@ -136,14 +140,14 @@
       </div>
 
       <div class="action desktop-only">
-        <ion-button @click="saveThreshold()">
+        <ion-button :disabled="isServiceScheduling" @click="saveThreshold()">
           <ion-icon slot="start" :icon="saveOutline" />
-          {{ $t("Save threshold rule") }}
+          {{ $t($route.query.id? "Update threshold rule" : "Save threshold rule") }}
```
Comment (ymaheshwari1) [inline src/views/SelectProduct.vue:145]:
```suggestion
          {{ $t($route.query.id ? "Update threshold rule" : "Save threshold rule") }}
```

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/SelectProduct.vue [hunk 7]
```diff
@@ -11,6 +12,9 @@
           <ion-button fill="clear" class="mobile-only">
             <ion-icon :icon="filterOutline" />
           </ion-button>
+          <ion-button v-if="isFilterChanged" fill="clear">
+            <ion-icon slot="end" color="warning" :icon="warningOutline"></ion-icon>
```
Comment (ymaheshwari1) [inline src/views/SelectProduct.vue:16]: Use self-closing tag.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/store/modules/job/getters.ts [hunk 8]
```diff
@@ -39,21 +40,22 @@ const getters: GetterTree <JobState, RootState> = {
...
-    getTagsIncluded: (state) => (id: string): any => {
+    getTagsAndOperator: (state, getters) => (id: string, type: string): any => {
+      const tagString = getters.getTags(id, type);
+      if(tagString){
+        let tags = parser.removeOuterBrackets(tagString.trim());
+        const operator = tagString.indexOf(' AND ') > 0 ? 'AND' : 'OR'
+        tags = tags.split(` ${operator} `).map((tag: any) => JSON.parse(tag))
+        return { tags, operator }
+      } 
+      return "";
```
Comment (adityasharma7) [inline src/store/modules/job/getters.ts]: I think we should return an object

File: src/store/modules/job/getters.ts [hunk 9]
```diff
@@ -39,21 +40,22 @@ const getters: GetterTree <JobState, RootState> = {
     getJobs: (state) => {
       return state.cached;
     },
-    getTagsIncluded: (state) => (id: string): any => {
+    getTagsAndOperator: (state, getters) => (id: string, type: string): any => {
+      const tagString = getters.getTags(id, type);
+      if(tagString){
+        let tags = parser.removeOuterBrackets(tagString.trim());
+        const operator = tagString.indexOf(' AND ') > 0 ? 'AND' : 'OR'
+        tags = tags.split(` ${operator} `).map((tag: any) => JSON.parse(tag))
```
Comment (adityasharma7) [inline src/store/modules/job/getters.ts:48]: Add comment why we need to parse

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/views/SelectProduct.vue [hunk 10]
```diff
@@ -264,15 +345,109 @@ export default defineComponent({
...
+            let shopifyConfigId = this.shopifyConfig[productStoreId]
+            let facilityId = this.facilitiesByProductStore[productStoreId]
+            if(!shopifyConfigId) {
+              const shopifyConfig = await this.store.dispatch('util/getShopifyConfig', productStoreId)
+              shopifyConfigId = shopifyConfig.shopifyConfigId
+            }
 
-      saveThresholdModal.present();
+            if (!facilityId) {
+              const resp = await this.store.dispatch('util/fetchFacilitiesByProductStore', {
```
Comment (adityasharma7) [inline src/views/SelectProduct.vue:386]: I think we should set these values as is from job.

File: src/views/SelectProduct.vue [hunk 11]
```diff
@@ -264,15 +345,109 @@ export default defineComponent({
...
+                fieldList: ['facilityId', 'productStoreId'],
+                distinct: 'Y',
+                noConditionFind: 'Y',
+                filterByDate: 'Y',
+                viewSize: 10
+              })
+              facilityId = resp[productStoreId]
+            }
+            const payload = {
+              'JOB_NAME': this.job.jobName ? this.job.jobName : this.userProfile.partyName,
```
Comment (adityasharma7) [inline src/views/SelectProduct.vue:401]: I think this value should be as is from current job

File: src/views/SelectProduct.vue [hunk 12]
```diff
@@ -264,15 +345,109 @@ export default defineComponent({
...
+              'SERVICE_NAME': this.job.serviceName,
+              'SERVICE_COUNT': '0',
+              'jobFields': {
+                'productStoreId': productStoreId,
+                'systemJobEnumId': this.job.systemJobEnumId,
+                'maxRecurrenceCount': '-1',
+                'parentJobId': this.job.parentJobId,
+                'recurrenceTimeZone': this.userProfile?.userTimeZone
```
Comment (adityasharma7) [inline src/views/SelectProduct.vue:409]: This should be same from the current job

File: src/views/SelectProduct.vue [hunk 13]
```diff
@@ -264,15 +345,109 @@ export default defineComponent({
...
+                viewSize: 10
+              })
+              facilityId = resp[productStoreId]
+            }
+            const payload = {
+              'JOB_NAME': this.job.jobName ? this.job.jobName : this.userProfile.partyName,
+              'SERVICE_NAME': this.job.serviceName,
+              'SERVICE_COUNT': '0',
+              'jobFields': {
+                'productStoreId': productStoreId,
```
Comment (adityasharma7) [inline src/views/SelectProduct.vue:405]: This should be same from the current job

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/services/JobService.ts [hunk 14]
```diff
@@ -1,4 +1,37 @@
...
+const fetchJob = async (payload: any): Promise<any> => {
+  let resp;
+  const params = {
+    "inputFields": {
+      "productStoreId": payload.eComStoreId,
+      "productStoreId_op": "equals",
+      "jobId": payload.jobId.toString(),
+      "jobId_op": "equals"
+    },
+    "fieldList": [ "systemJobEnumId", "runTime", "tempExprId", "parentJobId", "serviceName", "jobId", "jobName", "statusId", "cancelDateTime", "finishDateTime", "startDateTime", "runtimeDataId", "productStoreId" ],
```
Comment (ymaheshwari1) [inline src/services/JobService.ts:14]: Add `runtimeDate` in the list.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/views/SelectProduct.vue [hunk 15]
```diff
@@ -222,20 +240,89 @@ export default defineComponent({
...
+      isFilterChanged: false,
+      isServiceScheduling: false,
+      job: {} as any,
+      jobId: "" as any
+    }
+  },
+  async ionViewWillEnter(){
+    if (this.$route.query.id) {
+      this.jobId = this.$route.query.id;
+      const job = this.pendingJobs.find((job: any) => job.jobId === this.jobId) ? this.pendingJobs.find((job: any) => job.jobId === this.jobId) : await JobService.fetchJob({eComStoreId: this.getCurrentEComStore.productStoreId, jobId: this.jobId})
```
Comment (ymaheshwari1) [inline src/views/SelectProduct.vue:266]: Add this in a variable and then use.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

**Review by ymaheshwari1 — APPROVED** [review]

File: src/views/SelectProduct.vue [hunk 16]
```diff
@@ -222,20 +240,90 @@ export default defineComponent({
...
+      queryString: '',
+      isFilterChanged: false,
+      isServiceScheduling: false,
+      job: {} as any,
+      jobId: "" as any
+    }
+  },
+  async ionViewWillEnter(){
+    if (this.$route.query.id) {
+      this.jobId = this.$route.query.id;
```
Comment (adityasharma7) [inline src/views/SelectProduct.vue:265]: I think we should have method for it

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/views/SelectProduct.vue [hunk 17]
```diff
@@ -222,20 +240,90 @@ export default defineComponent({
...
+      let job = this.pendingJobs.find((job: any) => job.jobId === this.jobId)
+      job = job ? job : await JobService.fetchJob({eComStoreId: this.getCurrentEComStore.productStoreId, jobId: this.jobId})
+      if (job) {
+        this.job = job;
+        if (job.runtimeData?.searchPreferenceId) {
+          const includedTags = this.getTagsAndOperator(job.runtimeData.searchPreferenceId, "included").tags
```
Comment (adityasharma7) [inline src/views/SelectProduct.vue:271]: Store `this.getTagsAndOperator(job.runtimeData.searchPreferenceId, "included")` in a variable and then use

**Review by adityasharma7 — APPROVED** [review]

File: src/views/SelectProduct.vue [hunk 18]
```diff
@@ -222,20 +240,93 @@ export default defineComponent({
...
   },
   methods: {
+    async applyThresholdRule(id: any){
+      this.jobId = id;
+      let job = this.pendingJobs.find((job: any) => job.jobId === this.jobId)
+      job = job ? job : await JobService.fetchJob({eComStoreId: this.getCurrentEComStore.productStoreId, jobId: this.jobId})
+      if (job) {
+        this.job = job;
+        if (job.runtimeData?.searchPreferenceId) {
+          const includedTagsAndOperator = this.getTagsAndOperator(job.runtimeData.searchPreferenceId, "included");
```
Comment (adityasharma7) [inline src/views/SelectProduct.vue:276]: I think we should created a method that takes search preference and returns the appliedFilters object

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**